pkg/controller/infrastructure/infraflow/access/networking_access.go

```diff
@@ -253,23 +254,26 @@ func (a *networkingAccess) GetRouterInterfacePortID(routerID, subnetID string) (
 
 // RemoveRouterInterfaceAndWait removes the router interface. Either subnetID or portID must be specified
 func (a *networkingAccess) RemoveRouterInterfaceAndWait(ctx context.Context, routerID, subnetID, portID string) error {
+	var err error
+	ticker := time.Tick(3 * time.Second)
 	for {
-		if ctx.Err() != nil {
-			return ctx.Err()
-		}
-		_, err := a.networking.RemoveRouterInterface(routerID, routers.RemoveInterfaceOpts{SubnetID: subnetID, PortID: portID})
-		if err != nil {
-			if _, ok := err.(gophercloud.ErrDefault404); ok {
-				return nil
+		select {
+		case <-ctx.Done():
+			return fmt.Errorf("%w, last error: %w", ctx.Err(), err)
+		case <-ticker:
+			_, err := a.networking.RemoveRouterInterface(routerID, routers.RemoveInterfaceOpts{SubnetID: subnetID, PortID: portID})
+			if err != nil {
+				if _, ok := err.(gophercloud.ErrDefault404); ok {
+					return nil
+				}
+				if _, ok := err.(gophercloud.ErrDefault409); !ok {
+					return err
+				}
 			}
-			if _, ok := err.(gophercloud.ErrDefault409); !ok {
-				return err
+			if err == nil {
+				return nil
```

Comment:
Does this look better:

```
				if _, ok := err.(gophercloud.ErrDefault409); !ok {
					return err
				}
                continue
            }
            return nil
```

It's not entirely on this PR for the style change but it looks more readable in my opinion